CDCgov/prime-reportstream
# Adoption of React Router v6

This PR updates our dependency version of `react-router` from major version 5 to major version 6. Additionally, the following refactors were made to either solve deprecation errors:

- Documentation added for all new patterns
- App Router extracted from App.tsx
- Deprecation of AuthRoute, now AuthElement pattern
- Updates the IA framework routing
- Deprecates path matching for useParams pattern
- Adds useValidParams to check params aren't undefined and throw if they are
- Deprecates useHistory for useNavigate hook
- Updates tests for all broken by router update
- Adds tests for new elements and functions

Test Steps:
1. Pull latest
2. Do not log in -- check the public site for routing errors.
3. Log in -- check Daily Data, Submissions, and Admin pages for routing errors or uncaught errors

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #5004 
- Fixes #6694

## Continued work

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
